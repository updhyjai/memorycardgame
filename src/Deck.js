import React from 'react';
import './Game.css';

function Deck(props) {
    
    const handleCardClick= (e, code, deck)=>{
        props.onCardClick(e,code,deck);
    }
    let deck = props.cards.map((card,index) => {
        return (
          <div
            className="cards"
            onClick={(e) => handleCardClick(e, card.code, props.deckId)}
            key={index + card.code}
          >
            <img alt = {card.code} src={card.image} style = {{visibility:"hidden"}}></img>
          </div>
        );
      });
  return (
   deck
  );
}

export default Deck;
