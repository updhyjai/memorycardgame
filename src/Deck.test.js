import React from 'react';
import Deck from './Deck';
import renderer from 'react-test-renderer';

describe('The deck of cards',()=>{
    it('display list of cards',()=>{
        const cards =  [{"value": "6",
        "code": "6C", "images": {"svg": "https://deckofcardsapi.com/static/img/6C.svg", "png": "https://deckofcardsapi.com/static/img/6C.png"}, "suit": "CLUBS", "image": "https://deckofcardsapi.com/static/img/6C.png"}, {"value": "9", "code": "9D", "images": {"svg": "https://deckofcardsapi.com/static/img/9D.svg", "png": "https://deckofcardsapi.com/static/img/9D.png"}, "suit": "DIAMONDS", "image": "https://deckofcardsapi.com/static/img/9D.png"}, {"value": "10", "code": "0S", "images": {"svg": "https://deckofcardsapi.com/static/img/0S.svg", "png": "https://deckofcardsapi.com/static/img/0S.png"}, "suit": "SPADES", "image": "https://deckofcardsapi.com/static/img/0S.png"}, {"value": "2", "code": "2C", "images": {"svg": "https://deckofcardsapi.com/static/img/2C.svg", "png":
        "https://deckofcardsapi.com/static/img/2C.png"}, "suit": "CLUBS", "image": "https://deckofcardsapi.com/static/img/2C.png"}, {"value": "7", "code": "7H", "images": {"svg": "https://deckofcardsapi.com/static/img/7H.svg", "png": "https://deckofcardsapi.com/static/img/7H.png"}, "suit": "HEARTS", "image": "https://deckofcardsapi.com/static/img/7H.png"}, {"value": "QUEEN", "code": "QH", "images": {"svg": "https://deckofcardsapi.com/static/img/QH.svg", "png": "https://deckofcardsapi.com/static/img/QH.png"}, "suit": "HEARTS", "image": "https://deckofcardsapi.com/static/img/QH.png"}, {"value": "ACE", "code": "AS", "images": {"svg": "https://deckofcardsapi.com/static/img/AS.svg", "png": "https://deckofcardsapi.com/static/img/AS.png"}, "suit": "SPADES", "image": "https://deckofcardsapi.com/static/img/AS.png"}, {"value": "JACK", "code": "JD", "images": {"svg": "https://deckofcardsapi.com/static/img/JD.svg", "png": "https://deckofcardsapi.com/static/img/JD.png"}, "suit": "DIAMONDS", "image": "https://deckofcardsapi.com/static/img/JD.png"}, {"value": "7", "code": "7D", "images": {"svg": "https://deckofcardsapi.com/static/img/7D.svg", "png": "https://deckofcardsapi.com/static/img/7D.png"}, "suit": "DIAMONDS", "image": "https://deckofcardsapi.com/static/img/7D.png"}]
        const tree = renderer
                        .create(<Deck cards = {cards} />)
                        .toJSON();
         console.log(tree);      
         expect(tree).toMatchSnapshot();         
    })
})