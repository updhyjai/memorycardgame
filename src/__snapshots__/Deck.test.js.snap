// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The deck of cards display list of cards 1`] = `
Array [
  <div
    className="cards"
    onClick={[Function]}
  >
    <img
      alt="6C"
      src="https://deckofcardsapi.com/static/img/6C.png"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </div>,
  <div
    className="cards"
    onClick={[Function]}
  >
    <img
      alt="9D"
      src="https://deckofcardsapi.com/static/img/9D.png"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </div>,
  <div
    className="cards"
    onClick={[Function]}
  >
    <img
      alt="0S"
      src="https://deckofcardsapi.com/static/img/0S.png"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </div>,
  <div
    className="cards"
    onClick={[Function]}
  >
    <img
      alt="2C"
      src="https://deckofcardsapi.com/static/img/2C.png"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </div>,
  <div
    className="cards"
    onClick={[Function]}
  >
    <img
      alt="7H"
      src="https://deckofcardsapi.com/static/img/7H.png"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </div>,
  <div
    className="cards"
    onClick={[Function]}
  >
    <img
      alt="QH"
      src="https://deckofcardsapi.com/static/img/QH.png"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </div>,
  <div
    className="cards"
    onClick={[Function]}
  >
    <img
      alt="AS"
      src="https://deckofcardsapi.com/static/img/AS.png"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </div>,
  <div
    className="cards"
    onClick={[Function]}
  >
    <img
      alt="JD"
      src="https://deckofcardsapi.com/static/img/JD.png"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </div>,
  <div
    className="cards"
    onClick={[Function]}
  >
    <img
      alt="7D"
      src="https://deckofcardsapi.com/static/img/7D.png"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </div>,
]
`;
